kong/llm/proxy/handler.lua

```diff
@@ -477,4 +484,70 @@ function _M:access(conf)
 
 end
 
+
+
+function _M:build_http2_alpn_filter(plugin_name)
+  -- do not execute if the kong configuration doesn't have any http2 listeners
+  local http2_enabled = false
+  for _, listener in ipairs(kong.configuration.proxy_listeners) do
+    if listener.http2 then
+      http2_enabled = true
+      break
+    end
+  end
+
+  if not http2_enabled then
+    ngx.log(ngx.INFO, "no http2 listeners found, skipping LLM plugin initialization")
+    return
+  end
+
+  local sni_cache_key = "ai:llm:cert_enabled_snis:" .. plugin_name
+  local orig_ssl_client_hello = Kong.ssl_client_hello   -- luacheck: ignore
+  Kong.ssl_client_hello = function()                   -- luacheck: ignore
+    orig_ssl_client_hello()
+
+    local ctx = ngx.ctx
+    -- ensure phases are set
+    ctx.KONG_PHASE = PHASES.client_hello
+
+    kong_global.set_namespaced_log(kong, plugin_name)
+    local snis_set, err = kong.cache:get(sni_cache_key, TTL_FOREVER,
+      sni_filter.build_ssl_route_filter_set, plugin_name)
```

Comment:
I wonder if this could cause similar performance impact, just the same as the mTLS plugin's case. Suggest we introduce this PR after https://github.com/Kong/kong-ee/pull/10510/files

Reply:
Let's merge the current PR first,it seems that's different functionality in here. When the ee PR merged, I will cherry-pick it to CE ( actually can not directly cherry-pick due to ee and ce db related core code difference)